### ISSUE TITLE
srb init is slow, takes 15 minutes

### ISSUE THREAD

**Issue by jaredbeck** [sorbet/sorbet#3753]
I run `srb init` a few times a week, when I update sorbet, per the [instructions](https://sorbet.org/docs/faq#how-do-i-upgrade-sorbet), so a 15-minute `init` is something I'd like to help improve.

#### Observed output

```
time bin/srb init
...
Generating: sorbet/rbi/todo.rbi
# <---- most time spent here, perhaps 10 min
✅ Done!
real	14m41.643s
user	25m11.435s
sys	0m19.992s
bin/srb --version
Sorbet typechecker 0.5.6124 git 90bb7d4470a74aeae60fe30a36de8412aa809f7c debug_symbols=true clean=1
```

#### Expected behavior

I feel like it used to be a lot faster? Sorry I can't be more specific.

----

This is a 50kLoC project with a 158-gem bundle. `init` takes 15 minutes, uses 200% CPU, and 4 GB RAM

What other information can I provide that might help?

**Comment by jez** [comment]
(sorry misclick closed the issue)

this has been slow on stripe's codebase in the past too. the only way i know how to debug this is to add print statements to the source code listing timings. for example, find the place where srb init requires all gems, and print a before / after timer for that gem, and the same for where it requires all files in a project

**Comment by jaredbeck** [comment]
Using `Benchmark.bm` in `Sorbet::Private::Main.init`, it seems that `SuggestTyped` is the culprit.

```
                                                               user     system      total        real
make_step(Sorbet::Private::CreateConfig).call              0.000069   0.000030   0.000099 (  0.000093)
make_step(Sorbet::Private::FetchRBIs).call                 0.009912   0.025246   0.104471 (  0.523368)
make_step(Sorbet::Private::GemGeneratorTracepoint).call   11.925064   3.142961  17.541126 ( 21.244406)
make_step(Sorbet::Private::HiddenMethodFinder).call      154.799889   3.013018 163.623398 (164.455781)
make_step(Sorbet::Private::TodoRBI).call                   0.000455   0.020348   3.508349 (  1.798987)
Sorbet::Private::SuggestTyped.suggest_typed                0.000625   0.002502 651.174080 (335.809089)
Sorbet::Private::TodoRBI.main                              0.002748   0.003624   3.354566 (  1.752041)
make_step(Sorbet::Private::SuggestTyped).call              0.003649   0.006374 1311.944993 (672.330729)
```

**Comment by jaredbeck** [comment]
It looks like `SuggestTyped` shells out to `srb tc --suggest-typed`. I'm not a C programmer, but it looks like there's a verbosity setting, `-vvv` so let's see if we can learn anything from that?

```
bin/srb tc --suggest-typed --error-white-list=7022 --typed=strict -a -vvv &> srb_tc.out &
egrep -v wait_pop_timed srb_tc.out > srb_tc.out.2
# ... lots of typecheckOne: 1ms
[T7362355][2020-12-23T20:15:53.109331] typecheckOne: 1ms
[T7362352][2020-12-23T20:15:53.111860] typecheckOne: 1ms
[T7362351][2020-12-23T20:15:53.111965] typecheckOne: 1ms
[T7362350][2020-12-23T20:15:53.116468] findAllReadsAndWrites: 2ms
[T7362350][2020-12-23T20:15:53.116492] cfg: 2ms
[T7362356][2020-12-23T20:15:53.116509] validateSymbols: 2ms
[T7362350][2020-12-23T20:15:53.116982] typecheckOne: 2ms
# ... this part is all wait_pop_timed, every 20ms
[T7362353][2020-12-23T20:21:07.141741] Inference::run: 314032ms
[T7362353][2020-12-23T20:21:07.142177] typecheckOne: 314032ms
[T7362354][2020-12-23T20:21:09.399526] Inference::run: 316320ms
[T7362354][2020-12-23T20:21:09.400260] Inference::run: 1ms
[T7362354][2020-12-23T20:21:09.400911] typecheckOne: 316323ms
# wooo! this is an outlier, for sure
[T7362342][2020-12-23T20:21:09.400978] Multiplexing job
[T7362350][2020-12-23T20:21:09.400999] Worker got task
```

But, like I said, I'm not a C programmer, so this is probably as far as I can get?

**Comment by jaredbeck** [comment]
With Dmitry's help, I used `--web-trace-file` to determine which methods the 5-minute `Inference::run`s represent. I added the `ignore` sigil to these files and my `srb init` time was reduced from 45 minutes to three minutes.

@dmitry-stripe I'm satisfied with this workaround, so I don't plan to look into this any further. If you'd like to reproduce this issue, I can send you the identified functions, redacted.
